src/legends/swatches.js

```diff
@@ -170,18 +173,22 @@ function legendItems(scale, options = {}, swatch, swatchStyle) {
       div.insert("style", "*").text(`
         .${className} {
           font-family: system-ui, sans-serif;
-          font-size: 10px;
-          margin-bottom: 0.5em;${
-            marginLeft === undefined
-              ? ""
-              : `
-          margin-left: ${+marginLeft}px;`
-          }${
-        width === undefined
+          font-size: 10px;${marginTop === undefined
           ? ""
-          : `
-          width: ${width}px;`
-      }
+          : `margin-top: ${+marginTop}px;`
+        }${marginRight === undefined
+          ? ""
+          : `margin-right: ${+marginRight}px;`
+        }${marginBottom === undefined
+          ? `margin-bottom: 0.5em;`
+          : `margin-bottom: ${+marginBottom}px;`
```

Comment:
this mix seems like a problem, not sure how to unify that properly. Should we default to the equivalent number of pixels, on line 93?

Reply:
I've updated it to default to `8px` which should be the equivalent. What do you think?